test/parallel/test-temporal.js

```diff
@@ -0,0 +1,14 @@
+// Flags: --harmony-temporal
+
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+if (!process.config.variables.v8_enable_temporal_support) {
+  common.skip('Temporal is not enabled');
+}
+
+// Use globalThis.Temporal to workaround linter complaints.
+assert.strictEqual(typeof globalThis.Temporal, 'object');
+const instant = globalThis.Temporal.Instant.from('1969-07-20T20:17Z');
+assert.strictEqual(instant.toString(), '1969-07-20T20:17:00Z');
```

Comment:
Wouldn't timezone support require ICU? If so, we should either skip the test when no ICU, or use a part of the API that doesn't involve timezone

Reply:
It seems that this PR adds a lot of timezone data as JSON files. Maybe ICU isn't used at all

Reply:
Updated the test to detect ICU support.

I think when there is no built-in ICU, the zoneinfo64 data can be directly bundled to fix the build error. I will follow up with this.